linkedin.py

```diff
@@ -161,74 +161,92 @@ def processJob(self, jobID: str, jobCounter: models.JobCounter):
-                    continue
-            
-            if jobTitle is None:
-                utils.logDebugMessage("Couldn't find jobTitle", utils.MessageTypes.WARNING)
+            companyName = self.getCompanyNameFromJobCard(jobItem)
+            if not companyName:
+                utils.logDebugMessage("Could not extract company name from job card", MessageTypes.WARNING)
                 continue
-            
-            if jobTitle is None:
-                utils.logDebugMessage("Couldn't find jobTitle", utils.MessageTypes.WARNING)
+
+            if self.isCompanyBlacklisted(companyName):
+                utils.logDebugMessage(f"Not adding job as company '{companyName}' is blacklisted", MessageTypes.INFO)
                 continue
-            
-            jobId = jobItem.get_attribute("data-occludable-job-id")
 
-            if jobId is None:
-                utils.logDebugMessage("Couldn't find jobID", utils.MessageTypes.WARNING)
+            jobId = jobItem.get_attribute("data-occludable-job-id")
+            if not jobId:
+                utils.logDebugMessage("Could not extract job ID from job card", MessageTypes.WARNING)
                 continue
 
+            workPlaceType = self.getWorkplaceTypeFromJobCard(jobItem)
+
             jobsForVerification.append(models.JobForVerification(
                 linkedinJobId=jobId.split(":")[-1],
                 title=jobTitle,
                 company=companyName,
                 workplaceType=workPlaceType))
 
         return jobsForVerification
-    
+
+
+    def getCompanyNameFromJobCard(self, jobItem) -> Optional[str]:
+        """Extract company name from a job card using LinkedIn's current structure"""
+        selectors = [
+            constants.JOB_CARD_COMPANY_NAME_CSS,
+            constants.JOB_CARD_SUBTITLE_CSS, 
+            constants.JOB_CARD_METADATA_CSS,
+            constants.JOB_CARD_COMPANY_CSS
+        ]
+        
+        for selector in selectors:
+            elements = jobItem.find_elements(By.CSS_SELECTOR, selector)
+            if elements and len(elements) > 0:
+                return utils.removeSeparators(elements[0].text)
+            
+        return None
+
+
+    def getJobTitleFromJobCard(self, jobItem) -> Optional[str]:
+        """Extract job title from a job card using LinkedIn's current structure"""
+        selectors = [
+            constants.JOB_CARD_TITLE_LINK_CSS,
+            constants.JOB_CARD_TITLE_HEADING_CSS,
+            constants.JOB_CARD_BASE_TITLE_CSS,
+            constants.JOB_CARD_TITLE_LABEL_CSS
+        ]
+        
+        for selector in selectors:
+            elements = jobItem.find_elements(By.CSS_SELECTOR, selector)
+            if elements and len(elements) > 0:
+                return elements[0].text.strip()
+            
+        return None
+
+
+    def getWorkplaceTypeFromJobCard(self, jobItem) -> str:
+        """Extract workplace type from a job card if available"""
```

Comment:
Instead, renaming the function to getWorkplaceTypeFromJobCardIfAvailable would make sense